src/pages-and-resources/discussions/app-config-form/apps/shared/DivisionByGroupFields.jsx

```diff
@@ -1,21 +1,75 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Form, TransitionReplace } from '@edx/paragon';
+import { FieldArray, useFormikContext } from 'formik';
 import FormSwitchGroup from '../../../../../generic/FormSwitchGroup';
 import messages from './messages';
 import AppConfigFormDivider from './AppConfigFormDivider';
+import { updateDividedCourseWideDiscussionsIds } from '../../../data/slice';
+import { updateAppConfigs } from '../../../data/thunks';
 
 function DivisionByGroupFields({
-  onBlur,
-  onChange,
-  intl,
-  values,
+  onBlur, onChange, intl, values,
 }) {
+  const dispatch = useDispatch();
+  const { setFieldValue } = useFormikContext();
+
+  useEffect(() => {
+    dispatch(updateDividedCourseWideDiscussionsIds(values.dividedCourseWideDiscussionsIds));
+  }, [values.dividedCourseWideDiscussionsIds]);
+
+  useEffect(() => {
+    const discussionTopicIds = values.discussionTopics?.map(
```

Comment:
can we destructure **dividedCourseWideDiscussionsIds** or similar keys from values to avoid writing values.key every time we use in code.?

Reply:
Yes we can destructure it at top of the file. As we are also performing destructuring on DivisionByGroupFields#45 for update the appConfig model so we need to assign different name alias. like 
const { dividedCourseWideDiscussionsIds: CourseWideDiscussionsIds } = appConfig;